insertion_sort.go

```diff
@@ -2,32 +2,149 @@ package main
 
 import (
 	"fmt"
-
-	"github.com/prog-1/list"
 )
 
-// LessFunc returns true if a is smaller than b.
+type T int
```

Comment:
Removing "github.com/prog-1/list" import and copying everything here is wrong. You should have used the imported module.